media/src/androidMain/kotlin/dev/icerock/moko/media/picker/MediaPickerControllerImpl.kt

```diff
@@ -51,90 +66,93 @@ internal class MediaPickerControllerImpl(
 
-        val currentFragment: Fragment? = fragmentManager.findFragmentByTag(mediaPickerFragmentTag)
-        val pickerFragment: MediaPickerFragment = if (currentFragment != null) {
-            currentFragment as MediaPickerFragment
-        } else {
-            MediaPickerFragment().apply {
-                fragmentManager
-                    .beginTransaction()
-                    .add(this, mediaPickerFragmentTag)
-                    .commitNow()
-            }
-        }
-
         return suspendCoroutine { continuation ->
             val action: (Result<Media>) -> Unit = { continuation.resumeWith(it) }
-            pickerFragment.pickMedia(action)
+            mediaPickerDelegate.pickMedia(action)
         }
     }
 
     override suspend fun pickFiles(): FileMedia {
-        val fragmentManager =
-            fragmentManager ?: error("can't pick image without active window")
-
         permissionsController.providePermission(Permission.STORAGE)
 
-        val currentFragment: Fragment? = fragmentManager.findFragmentByTag(filePickerFragmentTag)
-        val pickerFragment: FilePickerFragment = if (currentFragment != null) {
-            currentFragment as FilePickerFragment
-        } else {
-            FilePickerFragment().apply {
-                fragmentManager
-                    .beginTransaction()
-                    .add(this, filePickerFragmentTag)
-                    .commitNow()
-            }
-        }
-
         val path = suspendCoroutine<FileMedia> { continuation ->
             val action: (Result<FileMedia>) -> Unit = { continuation.resumeWith(it) }
-            pickerFragment.pickFile(action)
+            filePickerDelegate.pickFile(action)
         }
 
         return path
     }
 
+    private suspend fun awaitActivity(): Activity {
+        val activity = activityHolder.value
+        if (activity != null) return activity
+
+        return withTimeoutOrNull(AWAIT_ACTIVITY_TIMEOUT_DURATION_MS) {
+            activityHolder.filterNotNull().first()
+        } ?: error(
+            "activity is null, `bind` function was never called," +
+                    " consider calling mediaPickerController.bind(activity)" +
+                    " or BindMediaPickerEffect(mediaPickerController) in the composable function," +
+                    " check the documentation for more info: " +
+                    "https://github.com/icerockdev/moko-media/blob/master/README.md"
+        )
+    }
+
     private fun MediaSource.requiredPermissions(): List<Permission> {
         return when (this) {
             MediaSource.GALLERY -> listOf(Permission.GALLERY)
             MediaSource.CAMERA -> listOf(Permission.CAMERA)
         }
     }
+
+    companion object {
+        private const val AWAIT_ACTIVITY_TIMEOUT_DURATION_MS = 2000L
+        private const val DEFAULT_FILE_NAME = "image.png"
+        private const val FILE_PROVIDER_SUFFIX = ".moko.media.provider"
```

Comment:
i dont see where this fileprovider declared in manifest. it missed?

Reply:
https://github.com/ZiXOps/moko-media/blob/replace-fragment-activity-usage/media/src/androidMain/AndroidManifest.xml#L7